src/torchcodec/share/cmake/TorchCodec/TorchCodecConfig.cmake

```diff
@@ -0,0 +1,185 @@
+# FindTorchCodec
+# --------------
+#
+# Finds the TorchCodec library
+#
+# This will define the following variables:
+#
+#   TORCHCODEC_FOUND        -- True if the system has the TorchCodec library
+#   TORCHCODEC_VARIANTS     -- List of TorchCodec variants
+#
+# and the following imported targets:
+#
+#   torchcodec::ffmpeg${N}
+#   torchcodec::core${N}
+#
+# where N is a TorchCodec variant from TORCHCODEC_VARIANTS list.
+
+include(FindPackageHandleStandardArgs)
+
+# Assume we are in <install-prefix>/share/cmake/TorchCodec/TorchCodecConfig.cmake
+get_filename_component(CMAKE_CURRENT_LIST_DIR "${CMAKE_CURRENT_LIST_FILE}" PATH)
+get_filename_component(TORCHCODEC_INSTALL_PREFIX "${CMAKE_CURRENT_LIST_DIR}/../../../" ABSOLUTE)
+
+# Include directories.
+set(TORCHCODEC_INCLUDE_DIRS ${TORCHCODEC_INSTALL_PREFIX}/_core)
+set(TORCHCODEC_VARIANTS "")
+
+function(add_ffmpeg_target ffmpeg_major_version libs)
+    set(target "torchcodec::ffmpeg${ffmpeg_major_version}")
+    set(prefix "TORCHCODEC_FFMPEG${ffmpeg_major_version}_INSTALL_PREFIX")
+    if (NOT DEFINED ENV{${prefix}})
+        message("Skipping ${target} as ${prefix} is not defined")
+        return()
+    endif()
+
+    set(prefix "$ENV{${prefix}}")
+    set(incdir "${prefix}/include")
+    if (UNIX AND NOT APPLE)
+        set(libdir "${prefix}/lib")
+    else()
+        message("Skipping ${target} on non-Linux platform")
+        return()
```

Comment:
Let's just asset this is `UNIX AND NOT APPLE` at the top of this file. It's OK to only support linux for now.